build.gradle

```diff
@@ -1987,28 +2005,30 @@ project(':connect:json') {
   archivesBaseName = "connect-json"
 
   dependencies {
-    compile project(':connect:api')
-    compile libs.jacksonDatabind
-    compile libs.jacksonJDK8Datatypes
-    compile libs.slf4jApi
+    implementation project(':connect:api')
```

Comment:
I think this is the correct mapping. 

By switching to `implementation`, any external project depending on the `connect:json` module will have to include an explicit dependency on `connect:api`. This appears to be have the same effect as what previous branches do with `compile` on earlier branches, but using `implementation` seems more correct since the [Gradle docs](https://docs.gradle.org/current/userguide/java_library_plugin.html) suggest that `compile` (or `compileOnly`) are more applicable for shading, which is not our case here.

Reply:
Does `connect-json` expose any `connect-api` types in its public API?

Reply:
The public API is defined in `connect:api`, and `connect:json` and `connect:transforms` are implementations of interfaces defined in (and therefore cannot be used without) `connect:api`. Within AK, we provide the `connect:json` and `connect:transforms` always with the `connect:api` (and runtime) modules, and therefore using `implementation` (and `compile` in earlier branches) seems appropriate.

From a more abstract perspective, one might argue that the `connect:json` and `connect:transforms` could declare their use of `connect:api` as a transitive dependency (e.g., `api` rather than `implementation`). It's just that as a practical matter we don't need this, and arguably no other projects would either.

As such, I think it's safer to use `implementation`, as it's in line with what we were doing earlier.

Reply:
To clarify, `compile` is more like `api` than `implementation`. I switched the dependency of `connect-api` for `connect-json` and `connect-transform` to `api` for now. We can change this in a future API, if we like, but it seems more consistent with the generally conservative approach we're taking in this PR.